docs/flank-github-action/flank_github_acton_sdd.md

```diff
@@ -0,0 +1,168 @@
+
+# Goals
+
+- Flank actions should be as easy as possible to run
+- Flank action should be maintainable together with Flank runner (always be up to date with options)
+- Flank action should be available on the public repository
+
+# Design
+
+There should be a possibility to specify all flank options using Github action variables as input, as well as passing the configuration itself.
+
+User must provide at least:
+
+- Flank version (default latest)
+- platform to run (iOS or Android)
+- Flank service account file to authenticate
+- Flank options or flank configuration file path
+
+# API
+
+Flank Github action will be developed using [composite run steps action](https://docs.github.com/en/actions/creating-actions/creating-a-composite-run-steps-action) which just packs other actions and runs it as a single action.
+
+### Design proposal
+
+```yaml
+name: 'Flank'
+description: 'Run Flank from Github actions!'
+inputs:
+	version: 
+		description: 'Version of flank to run'
+		required: true
+		default: <latest>
+	platform: 
+		description: 'Platform to run iOS or Android'
+		required: true
+  flank_option1:
+    description: 'Description of option 1'
+    required: false # validation will done when running
+    ...
+  flank_optionX:
+    description: 'Description of option X'
+    required: false # validation will done when running
+  flank_configuration_file:
+    description: 'Path to configuration file'
+    required: false # validation will done when running
+	flank_service_account: 
+	  description: 'Path to service account file to authenticate'
+    required: true
+outputs:
+  output_report:
+    description: "Output report"
+    value: ${{ steps.report.outputs.random-id }}
+runs:
+  using: "composite"
+  steps:
+    - id: download flank
+      run: <download flank>
+      shell: bash
+    - id: validate configuration
+      run: <check if configuation file or any options are specified>
+      shell: bash
+    - id: run flank
+      run: <check if configuation file or any options are specified>
+      shell: bash
+    - id: report
+      run: <show run report to user>
+      shell: bash
+```
+
+
+
+### Usage
+
+##### With config file
+
+```yaml
+on: [push]
+
+jobs:
+  hello_world_job:
+    runs-on: ubuntu-latest
+    name: A job to say hello
+    steps:
+    - uses: actions/checkout@v2
+    - id: flank
+      uses: actions/flank@v1
+      with:
+        version: '21.01.0'
+        platform: 'android'
+        flank_service_account: './service_account.json'
+        flank_configuration_file: './flank.yml'
+    - run: cat ${{ steps.flank.outputs.output_report }}
+      shell: bash
+```
+
+##### With options
+
+```yaml
+on: [push]
+
+jobs:
+  hello_world_job:
+    runs-on: ubuntu-latest
+    name: A job to say hello
+    steps:
+    - uses: actions/checkout@v2
+    - id: flank
+      uses: actions/flank@v1
+      with:
+        version: '21.01.0'
+        platform: 'android'
+        flank_service_account: './service_account.json'
```

Comment:
`service_account.json` - shound't be kept in secrets? I am not sure, just asking.

Reply:
it will be provided by the user, so it will be his responsibility to keep it safe

Reply:
But if it's file path user should add service acc to repo or create before this step yes? Maybe we could allow users to provide a string with service acc data instead of the file path.